src/main/java/luke/parser/Parser.java

```diff
@@ -0,0 +1,204 @@
+                }
+            }
+            return new InvalidCommand();
+        } catch (NumberFormatException e) {
+            return new InvalidCommand("The force cannot convert the value to a number.");
+        } catch (DateTimeParseException e) {
+            return new InvalidCommand("The force does not comprehend the date.");
+        } catch (IllegalArgumentException e) {
+            return new InvalidCommand(e.getMessage());
+        }
+    }
+
+    /**
+     * Prepares and return the appropriate read command.
+     *
+     * @param cmdAction The command action tied to the user input.
+     * @param args      The arguments that the user have supplied.
+     * @return The appropriate read command based on user input.
+     */
+    private static ReadCommand prepareReadCommand(CommandAction cmdAction, String[] args) {
+        switch (cmdAction) {
+        case FIND:
+            String keyword = parseFindArgument(args);
+            return new FindCommand(keyword);
+        default:
+            return new ListCommand();
+        }
+    }
+
+
+    /**
+     * Prepares and return the appropriate update command.
+     *
+     * @param cmdAction The command action tied to the user input.
+     * @param args      The arguments that the user have supplied.
+     * @return The appropriate update command based on user input.
+     */
+    private static UpdateCommand prepareUpdateCommand(CommandAction cmdAction, String[] args) {
+        int index = parseUpdateArguments(args);
+        switch (cmdAction) {
+        case MARK:
+            return new MarkCommand(index);
+        case UNMARK:
+            return new UnmarkCommand(index);
+        default:
+            return new DeleteCommand(index);
+        }
+    }
+
+    /**
+     * Prepares and return the appropriate add command.
+     *
+     * @param cmdAction The command action tied to the user input.
+     * @param args      The arguments that the user have supplied.
+     * @return The appropriate add command based on user input.
+     */
+    private static AddCommand prepareAddCommand(CommandAction cmdAction, String[] args) {
+        Map<String, String> argsMap = parseAddArguments(cmdAction, args);
+        switch (cmdAction) {
+        case DEADLINE:
+            return new DeadlineCommand(new Deadline(argsMap));
+        case EVENT:
+            return new EventCommand(new Event(argsMap));
+        default:
+            return new TodoCommand(new Todo(argsMap));
+        }
+    }
+
+
+    /**
+     * Parses and return the arguments tied to the add commands.
+     *
+     * @param cmdAction The command action tied to the user input.
+     * @param args      The arguments that the user have supplied.
+     * @return The arguments mapped to the keywords of the command.
+     * @throws IllegalArgumentException If the argument does not fit the syntax of the command.
+     */
+    private static Map<String, String> parseAddArguments(CommandAction cmdAction, String[] args)
```

Comment:
Really like how you've made use of hashmaps to parse the arguments from the user input. It's a pretty creative way to parse the user's argument! 👍